### DISCUSSION TITLE
Success stories from adopters of rules_js

### DISCUSSION THREAD

**Discussion by gregmagolan** [aspect-build/rules_js#1000] [Show and tell]
We are excited to help companies and developers build and test JavaScript at scale using Bazel. If you are currently using rules_js and would like to share your experiences or give a shout-out to the project, please post your company name and/or open source repository here. We look forward to hearing from you!

**Comment by Aghassi** [comment]
Robinhood is in the process of migrating to it. We currently have a limited use for libraries and are aiming to move apps over by end of year.

**Comment by arobinson** [comment]
[Nextworld](https://nextworld.net) is using rules_* for our Bazel Angular build. We needed a high performant build for our software and Angular CLI could not achieve the needed build times. Although `ngc` is still a bottleneck, using Bazel with the Aspect libraries, we were able to replace Angular CLI in our build getting at least 30% faster build times.

**Comment by dgp1130** [comment]
I know Greg is already familiar with it and probably tired of me asking for help, but I've been working on [`rules_prerender`](https://github.com/dgp1130/rules_prerender/) off and on in my spare time. The idea is a Bazel ruleset built on top of `rules_js` to provide a static site generator with Bazel. Basically, `bazel build //my_site` generates a directory containing prerendered HTML, JavaScript, CSS, and assets used in your application all bundled and laid out ready for production. Just upload to static file hosting!

`rules_prerender` is HTML-first, with no required client-side JS. It allows very fine grained control over the output, so you know _exactly_ what you're generating. It provides a flexible component model and bundles all the JavaScript and CSS for you. I've landed all the major tasks, so it's fairly feature complete at this point.

Unfortunately the biggest thing I haven't done is document it, so that's the next thing on my plate. If it sounds interesting, I'm happy to help anyone get started and play around with it to make up for the lack of documentation at the moment. And if prerendering HTML or static site generation with Bazel sound interesting to you, I'd love to chat and hear your thoughts. Please reach out!

**Comment by jhchabran** [comment]
[Sourcegraph](https://sourcegraph.com) has been migrating to Bazel for the last few months. Our CI pipeline has grown out of hands in terms of complexity over the years and lowering the build times was critical at this point. 

While on the backend side of things, we were able to work on our own, when it came to the client side, investing in commercial support from Aspect has been of a tremendous help. It would have taken much more time for us to get there without them and made the migration much less risky from an engineering and time investment perspective. 

At the time of writing this, the client code builds 2x times faster and we've reduced the CI minutes spent on these targets by a factor of 8.

And in the end, they also helped a lot on the backend side, Bazel is great, but sometimes, pinning down the exact source of errors is tedious and their support really shined there. 

Our code is fully public over https://github.com/sourcegraph/sourcegraph and you can browse the commits to see the process taken by Aspect to help us finally run Bazel in CI.

**Comment by ericrafalovsky** [comment]
[Asana](https://asana.com/) has been adopting rules_js / rules_ts which have significantly improved the maintainability of our Bazel monolith, compared to previous homegrown Bazel rulesets. Aspect has also been great to work with and has significantly accelerated many facets of this migration.

**Comment by jgao54** [comment]
Figma has been successfully using rules_ts / rules_js in production. Folks from Aspect have been great to work with when it comes to feature requests and bug fixes.

**Comment by pat-trunk-io** [comment]
Trunk.io has been using rules_js for many years now.
Bazel and rules_js allows our engineers to be productive from day one. Reproducible build are such an invaluable part of keeping our company moving fast.

The migration from 1.x to 2.0 was painless and are looking forward to the new features to come!

**Comment by mcevoypeter** [comment]
I'm using rules_js, rules_ts, rules_swc, and rules_format for a new TypeScript monorepo, and I've been pleased with how solid and well-structured the build process is. It has me excited to scale the project and put Bazel through its paces.
